controller/workitem_blackbox_test.go

```diff
@@ -101,6 +108,125 @@ func TestGetWorkItemWithLegacyDescription(t *testing.T) {
+
+	// clear the dataArray
+	dataArray = dataArray[:0]
+	dataArray = append(dataArray, result4.Data)
+	payload3 := minimumRequiredReorderPayload()
+	payload3.Data = dataArray
+	payload3.Position.ID = *result5.Data.ID
+	payload3.Position.Direction = below
+	_, reordered2 := test.ReorderWorkitemOK(t, nil, nil, controller, &payload3)
+	require.Len(t, reordered2.Data, 1)
+	assert.Equal(t, result4.Data.Attributes["version"].(int)+1, reordered2.Data[0].Attributes["version"])
+	assert.Equal(t, *result4.Data.ID, *reordered2.Data[0].ID)
+
+	//Order of reordered workitems should be greater than the order of workitem position.id when position.direction is below
+	assert.True(t, reordered2.Data[0].Attributes["order"].(float64) > result4.Data.Attributes["order"].(float64))
+}
+
+// TestReorderWorkitemBadRequest is negative test which tests unsuccessful reorder by providing invalid input
+func TestReorderWorkitemBadRequest(t *testing.T) {
+	resource.Require(t, resource.Database)
+	priv, _ := almtoken.ParsePrivateKey([]byte(almtoken.RSAPrivateKey))
+	svc := testsupport.ServiceAsUser("TestGetWorkItem-Service", almtoken.NewManagerWithPrivateKey(priv), testsupport.TestIdentity)
+	require.NotNil(t, svc)
+	controller := NewWorkitemController(svc, gormapplication.NewGormDB(DB))
+	require.NotNil(t, controller)
+	payload1 := minimumRequiredCreateWithType(workitem.SystemBug)
+	payload1.Data.Attributes[workitem.SystemTitle] = "Reorder Test WI"
+	payload1.Data.Attributes[workitem.SystemState] = workitem.SystemStateClosed
+	_, result1 := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload1)
+	defer test.DeleteWorkitemOK(t, nil, nil, controller, *result1.Data.ID)
+	payload2 := minimumRequiredReorderPayload()
+
+	// This case gives empty dataArray as input
+	// Response is Bad Parameter
+	// Reorder is unsuccessful
+
+	var dataArray []*app.WorkItem2
+	payload2.Data = dataArray
+	payload2.Position.ID = *result1.Data.ID
+	payload2.Position.Direction = above
+	test.ReorderWorkitemBadRequest(t, nil, nil, controller, &payload2)
+}
+
+// TestReorderWorkitemNotFound is negative test which tests unsuccessful reorder by providing invalid input
+func TestReorderWorkitemNotFound(t *testing.T) {
+	resource.Require(t, resource.Database)
+	priv, _ := almtoken.ParsePrivateKey([]byte(almtoken.RSAPrivateKey))
+	svc := testsupport.ServiceAsUser("TestGetWorkItem-Service", almtoken.NewManagerWithPrivateKey(priv), testsupport.TestIdentity)
+	require.NotNil(t, svc)
+	controller := NewWorkitemController(svc, gormapplication.NewGormDB(DB))
+	require.NotNil(t, controller)
+	payload1 := minimumRequiredCreateWithType(workitem.SystemBug)
+	payload1.Data.Attributes[workitem.SystemTitle] = "Reorder Test WI"
+	payload1.Data.Attributes[workitem.SystemState] = workitem.SystemStateClosed
+	_, result1 := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload1)
+	defer test.DeleteWorkitemOK(t, nil, nil, controller, *result1.Data.ID)
```

Comment:
@aslakknutsen I think it is because the test in not part of a suite but  sure. @DhritiShikhar you could have something like this

```go
db := gormapplication.NewGormDB(DB)
defer cleaner.DeleteCreatedEntities(db)
```